metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -167,6 +167,7 @@ async def run_func_check_upgrade(pipeline: Pipeline) -> None:
                 logger.debug(
                     f"timed out with {e} waiting for version stats to be computed... skipping ahead."
                 )
+        sys.exit(ret)
```

Comment:
why are we existing with the return code of the version stats method?

Reply:
This caused the ingestion to fail in case we failed to get version stats. It makes no sense for ingestion to fail if version stats were not calculated.

Reply:
not quite - the `run_func_check_upgrade` method runs the ingestion pipeline, and if the pipeline succeeded, prints out an upgrade suggestion if appropriate. `ret` is the return code of the pipeline run

Removing the `sys.exit(ret)` call in the previous PR caused a serious regression: the ingestion runs always produced exit code 0, even if they actually failed!!

Reply:
this is unrelated to stats